sormas-backend/src/main/java/de/symeda/sormas/backend/survey/SurveyFacadeEjb.java

```diff
@@ -82,6 +103,76 @@ public SurveyDto save(@Valid SurveyDto dto) {
 		return toDto(survey);
 	}
 
+	@Override
+	@RightsAllowed(UserRight._SURVEY_EDIT)
+	public void uploadDocumentTemplate(SurveyReferenceDto surveyRef, DocumentTemplateDto uploadedDocumentTemplate, byte[] fileContent)
+		throws DocumentTemplateException {
+		Survey existingSurvey = surveyRef.getUuid() != null ? surveyService.getByUuid(surveyRef.getUuid()) : null;
+
+		if (existingSurvey != null) {
+			List<DocumentTemplateDto> availableTemplates =
+				documentTemplateFacade.getAvailableTemplates(new DocumentTemplateCriteria(DocumentWorkflow.SURVEY_DOCUMENT, null, surveyRef));
+			if (!availableTemplates.isEmpty()) {
+				availableTemplates.forEach(docTemplate -> {
+					DocumentTemplate existingDocumentTemplate = documentTemplateService.getByUuid(docTemplate.getUuid());
+					documentTemplateService.deletePermanent(existingDocumentTemplate, DocumentWorkflow.SURVEY_DOCUMENT);
+				});
+			}
+		}
+		DocumentTemplateDto savedDocumentTemplate = documentTemplateFacade.saveDocumentTemplate(uploadedDocumentTemplate, fileContent);
+		DocumentTemplate byReferenceDto = documentTemplateService.getByReferenceDto(savedDocumentTemplate.toReference());
+		assert existingSurvey != null;
+		existingSurvey.setDocumentTemplate(byReferenceDto);
+		surveyService.ensurePersisted(existingSurvey);
+	}
+
+	public boolean validateSurveyDocumentTemplate(byte[] fileContent) throws DocumentTemplateException {
+		IXDocReport report = templateEngine.readXDocReport(new ByteArrayInputStream(fileContent));
```

Comment:
this should be a private method and called from the upload method

Reply:
changed to the requested location